src/app/games/siam/tests/siam.component.spec.ts

```diff
@@ -151,15 +142,119 @@ describe('SiamComponent', () => {
 
         // When making the piece exit the board
         // The the orientation of the piece does not have to be chosen
-        await testUtils.expectClickSuccess('#clickPiece_4_4');
-        const move: SiamMove = new SiamMove(4, 4, MGPOptional.of(Orthogonal.DOWN), Orthogonal.DOWN);
-        await testUtils.expectMoveSuccess('#chooseDirection_DOWN', move);
+        await testUtils.expectClickSuccess('#square_4_4');
+        const move: SiamMove = SiamMove.of(4, 4, MGPOptional.of(Orthogonal.DOWN), Orthogonal.DOWN).get();
+        await testUtils.expectMoveSuccess('#square_4_5', move);
     }));
     it('should toast when clicking as first click on an empty square', fakeAsync(async() => {
         // Given the initial board
         // When clicking on an empty piece
         // Then a toast should say it's forbidden
-        const reason: string = SiamFailure.MUST_INSERT_OR_CHOOSE_YOUR_PIECE();
-        await testUtils.expectClickFailure('#insertAt_2_1', reason);
+        const reason: string = RulesFailure.MUST_CHOOSE_PLAYER_PIECE();
+        await testUtils.expectClickFailure('#square_2_1', reason);
+    }));
+    it('should cancel move when player clicks on the board instead of an orientation arrow', fakeAsync(async() => {
+        // Given that the player must select an orientation arrow
+        await testUtils.expectClickSuccess('#piece_0_0');
+        await testUtils.expectClickSuccess('#square_0_0');
+        spyOn(testUtils.getComponent(), 'cancelMoveAttempt').and.callThrough();
+        // When clicking somewhere else on the board
+        await testUtils.expectClickSuccess('#square_2_2');
+        // Then it should cancel the move
+        expect(testUtils.getComponent().cancelMoveAttempt).toHaveBeenCalledOnceWith();
+    }));
+    it('should cancel move when player selects an invalid target for insertion (empty)', fakeAsync(async() => {
+        // Given that the player must select the target for a move
+        await testUtils.expectClickSuccess('#piece_0_0');
+        // When the player clicks on an empty target which would result in an impossible move
+        spyOn(testUtils.getComponent(), 'cancelMoveAttempt').and.callThrough();
+        await testUtils.expectClickSuccess('#square_1_1');
+        // Then the move should be canceled
+        expect(testUtils.getComponent().cancelMoveAttempt).toHaveBeenCalledOnceWith();
+    }));
+    it('should cancel move when player selects an invalid target for insertion (mountain)', fakeAsync(async() => {
+        // Given that the player must select the target for a move
```

Comment:
erk, pas les given that :D !

Reply:
haha faudrait des règles de linter pour les commentaires :D